packages/kbn-apm-config-loader/src/config.ts

```diff
@@ -295,7 +295,7 @@ export class ApmConfiguration {
     const { servicesOverrides, redactUsers, ...configFromKibanaConfig } =
       this.getConfigFromKibanaConfig();
     const configFromEnv = this.getConfigFromEnv(configFromKibanaConfig);
-    const config = merge({}, configFromKibanaConfig, configFromEnv);
+    const config = deepMerge({}, configFromKibanaConfig, configFromEnv);
```

Comment:
Kibana can read configuration from its own config or environment variables. However, the way the configuration is merged, the env vars **always** override the config file (as we've seen in the case of `globalLabels`).

To prevent having to "mess" with types (or use another dependency), perhaps handling the merge manually will work.
e.g.
```
const globalLabels = merge({}, configFromKibanaConfig.globalLabels, configFromEnv.globalLabels);
delete configFromKibanaConfig.globalLabels;
delete configFromEnv.globalLabels;
const config = merge({}, configFromKibanaConfig, configFromEnv, globalLabels);
```
I'll try this locally first.